data/sea/52-psv.h

```diff
@@ -107,7 +123,72 @@ static int INLINE psv_compare (const char *xs, size_t xs_size, const char *ys, s
     return xs_size - ys_size;
 }
 
-static ierror_loc_t psv_read_buffer (psv_state_t *s)
+/* Seek to a line where the entity changes, or EOF */
+static int INLINE psv_skip_to_next_entity (psv_state_t *s)
+{
+    const char  *end_ptr    = s->input_ptr + s->input_size;
+    const char  *line_ptr   = s->input_ptr;
+    const char  *n_ptr      = 0;
+
+    char   *entity_cur      = s->entity_cur;
+    size_t  entity_cur_size = s->entity_cur_size;
+
+    iint_t  fact_count      = s->fact_count;
+
+    for (;;) {
+        const size_t bytes_remaining = end_ptr - line_ptr;
+
+        n_ptr = memchr (line_ptr, '\n', bytes_remaining);
+
+        if (n_ptr == 0) {
+            s->input_remaining = bytes_remaining;
+            return 1;
+        }
+
+        const char  *entity_ptr = line_ptr;
+        const int    new_entity = psv_compare (entity_cur, entity_cur_size, entity_ptr, bytes_remaining);
```

Comment:
I believe entity_cur contains the pipe after the entity name, so this will still count an entity 'ABCD' after 'AB' as a new entity - does that sound right?
might be worth writing a test for that case, perhaps just a bash script that runs bench with the max count set to 2?